### ISSUE TITLE
[Firefox Extension Incompatibility] Zooming via mouse or touchpad in gcode viewer no longer works

### ISSUE THREAD

**Issue by KenLucke** [OctoPrint/OctoPrint#3594]
I was about to report this as a OP bug, but when I got down to the Browser Being Used part, I decided to check it in Safari (I normally use Firefox), and it went away.  

Checking further, it seems that it is an incompatibility with one single plugin in Firefox, called "NoProfile," which "Blocks functions used for psychological profiling and tracking," and which I had recently re-enabled  

Removing or disabling that extension in Firefox returns OctoPrint to its normal function.

I am only reporting it here so that you may note it for future reference if someone reports it as a bug.

The following details the incompatibility (I wouldn't go so far as to call it a "bug" on the OP side of things).

#### What were you doing?
On the Gcode Viewer tab, attempting to zoom in or out on the model using either the touchpad gestures or using the mouse scroll wheel (or key-mouse combo) no longer work, as did previously.  clicking "Center Viewport on Model" or "Zoom In on Model" function as previously (albeit with their limitations), and dragging the model around in the "viewport" works, but trying to actually manually zoom in or out no longer works.

1. Load a model so that it shows inn the gcode viewer
2. attempt to zoom in or out using the mouse or touchpad

All files are affected

#### What did you expect to happen?
That the zooming function worked as previously

#### What happened instead?
It didn't

#### Did the same happen when running OctoPrint in safe mode?
Yes.  Entirely repeatable in normal AND safe mode

#### Version of OctoPrint
1.4.0 on OctoPi 0.17.0

#### Operating System running OctoPrint
Mac OS X 10.15.5

#### Browser and version of browser, operating system running browser
Firefox v77.0.1 ONLY when running extension called "NoProfile"

**Comment by GitIssueBot** [comment]
Hi @KenLucke,

It looks like there is some **information missing** from your bug report that will be needed in order to solve the problem. Read the [Contribution Guidelines](https://github.com/foosel/OctoPrint/blob/master/CONTRIBUTING.md) which will provide you with a template to fill out here so that your bug report is ready to be investigated (I promise I'll go away then too!).

If you did not intend to report a bug but wanted to **request a feature or brain storm** about some kind of development, please take special note of the title format to use as described in the [Contribution Guidelines](https://github.com/foosel/OctoPrint/blob/master/CONTRIBUTING.md).

**Please do not abuse the bug tracker as a support forum** - that can be found at [discourse.octoprint.org](https://discourse.octoprint.org). Go there for any kind of issues with network connectivity, webcam functionality, printer detection or any other kind of such support requests or general questions.

Also **make sure you are at the right place** - this is the bug tracker of the official version of OctoPrint, not the Raspberry Pi image OctoPi nor any unbundled third party OctoPrint plugins or unofficial versions. Make sure too that you have **read through the [Frequently Asked Questions](http://faq.octoprint.org)** and searched the [**existing tickets**](https://github.com/foosel/OctoPrint/search?q=&ref=cmdform&type=Issues) for your problem - try multiple search terms please.

I'm marking this one now as needing some more information. Please understand that if you do not provide that information within the next two weeks (until 2020-06-21 22:50 UTC) I'll close this ticket so it doesn't clutter the bug tracker. This is nothing personal, so please just be considerate and help the maintainers solve this problem quickly by following the guidelines linked above. Remember, the less time the devs have to spend running after information on tickets, the more time they have to actually solve problems and add awesome new features. Thank you!

Best regards,
~ Your friendly GitIssueBot

PS: I'm just an automated script, not a human being, so don't expect any replies from me :) Your ticket is read by humans too, I'm just not one of them.

**Comment by foosel** [comment]
I took a look at the source of the extension in question, and it works by explicitly unbinding all drag and zoom events on all elements in a page. So ... yeah. This extension nukes some functionality critical to the proper working of the gcode viewer. If you want to use the viewer, don't use that extension.

Closing as - as you also noticed - it ain't an issue with OctoPrint but rather the quite draconian extension. If someone falls over this via google the information will still be there.
